From: Miaohe Lin <linmiaohe@huawei.com>
Subject: mm: vmscan: mark kswapd_run() and kswapd_stop() __meminit
Date: Tue, 6 Jun 2023 20:18:13 +0800

Add __meminit to kswapd_run() and kswapd_stop() to ensure they're default
to __init when memory hotplug is not enabled.

Link: https://lkml.kernel.org/r/20230606121813.242163-1-linmiaohe@huawei.com
Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
Acked-by: Yu Zhao <yuzhao@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/linux/swap.h |    4 ++--
 mm/vmscan.c          |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

--- a/include/linux/swap.h~mm-vmscan-mark-kswapd_run-and-kswapd_stop-__meminit
+++ a/include/linux/swap.h
@@ -460,8 +460,8 @@ static inline bool node_reclaim_enabled(
 void check_move_unevictable_folios(struct folio_batch *fbatch);
 void check_move_unevictable_pages(struct pagevec *pvec);
 
-extern void kswapd_run(int nid);
-extern void kswapd_stop(int nid);
+extern void __meminit kswapd_run(int nid);
+extern void __meminit kswapd_stop(int nid);
 
 #ifdef CONFIG_SWAP
 
--- a/mm/vmscan.c~mm-vmscan-mark-kswapd_run-and-kswapd_stop-__meminit
+++ a/mm/vmscan.c
@@ -7856,7 +7856,7 @@ unsigned long shrink_all_memory(unsigned
 /*
  * This kswapd start function will be called by init and node-hot-add.
  */
-void kswapd_run(int nid)
+void __meminit kswapd_run(int nid)
 {
 	pg_data_t *pgdat = NODE_DATA(nid);
 
@@ -7877,7 +7877,7 @@ void kswapd_run(int nid)
  * Called by memory hotplug when all memory in a node is offlined.  Caller must
  * be holding mem_hotplug_begin/done().
  */
-void kswapd_stop(int nid)
+void __meminit kswapd_stop(int nid)
 {
 	pg_data_t *pgdat = NODE_DATA(nid);
 	struct task_struct *kswapd;
_
